### PR TITLE
Rename chainID to blockchainID

### PR DESCRIPTION
**Pull request by geoff-vball**
## Why this should be merged
`chainID` usually refers to an ethereum chainID, which is set in the chain genesis. This is distinct from `blockchainID`, which uniquely identifies a subnet on Avalanche, and is the hash of the P-Chain tx that creates the blockchain. We have used `chainID` in many places that should be `blockchainID`, so this PR looks to disambiguate them.

Note: `chainID` always appears as a `*big.Int` or uint256, whereas `blockchainID` is a `[32]byte`.

Closes https://github.com/ava-labs/teleporter/issues/95

## How is this documented

### CODE REVIEW

File: contracts/src/Teleporter/ITeleporterMessenger.sol [hunk 1]
```diff
@@ -238,30 +238,30 @@ interface ITeleporterMessenger {
...
-        bytes32 destinationChainID,
+        bytes32 destinationBlockchainID,
         uint256 messageID
     ) external view returns (address, uint256);
 
     /**
      * @dev Gets the next message ID to be used for a given chain ID.
      * @return The next message ID to be used to send a message to the given chain ID.
      */
-    function getNextMessageID(bytes32 chainID) external view returns (uint256);
```
Comment (cam-schultz) [inline contracts/src/Teleporter/ITeleporterMessenger.sol:249]: Can you update this to `destinationBlockchainID` for consistency with the rest of the contract?

File: contracts/src/Teleporter/ITeleporterMessenger.sol [hunk 2]
```diff
@@ -238,30 +238,30 @@ interface ITeleporterMessenger {
...
+    function getNextMessageID(bytes32 blockchainID) external view returns (uint256);
 
     /**
      * @dev Gets the number of receipts that have been sent to the given destination chain ID.
      * @return Size of the given queue.
      */
     function getReceiptQueueSize(
-        bytes32 chainID
```
Comment (cam-schultz) [inline contracts/src/Teleporter/ITeleporterMessenger.sol:256]: Same comment here `blockchainID` -> `originBlockchainID`

File: contracts/src/Teleporter/ITeleporterMessenger.sol [hunk 3]
```diff
@@ -238,30 +238,30 @@ interface ITeleporterMessenger {
...
 
     /**
      * @dev Gets the next message ID to be used for a given chain ID.
      * @return The next message ID to be used to send a message to the given chain ID.
      */
-    function getNextMessageID(bytes32 chainID) external view returns (uint256);
+    function getNextMessageID(bytes32 blockchainID) external view returns (uint256);
 
     /**
      * @dev Gets the number of receipts that have been sent to the given destination chain ID.
```
Comment (cam-schultz) [inline contracts/src/Teleporter/ITeleporterMessenger.sol:252]:
```suggestion
     * @dev Gets the number of receipts that are waiting to be sent to the given origin chain ID.
```

File: contracts/src/Teleporter/ITeleporterMessenger.sol [hunk 4]
```diff
@@ -238,30 +238,30 @@ interface ITeleporterMessenger {
...
-        bytes32 chainID
+        bytes32 blockchainID
     ) external view returns (uint256);
 
     /**
      * @dev Gets the receipt at the given index in the queue for the given chain ID.
      * @return The receipt requested.
      */
     function getReceiptAtIndex(
-        bytes32 chainID,
```
Comment (cam-schultz) [inline contracts/src/Teleporter/ITeleporterMessenger.sol:264]: Same comment here `blockchainID` -> `originBlockchainID`

File: contracts/src/Teleporter/ITeleporterMessenger.sol
Comment (cam-schultz) [inline contracts/src/Teleporter/ITeleporterMessenger.sol:1]: Can you please make the same changes I suggested in `TeleporterMessenger.sol` as well?

**Review by cam-schultz — COMMENTED** [review]
Thanks for taking care or this. I left a few comments to clean up some of the naming while we're here.

**Review by bernard-avalabs — APPROVED** [review]
Other than the few minor comments that Cam made, the PR LGTM.

**Review by cam-schultz — APPROVED** [review]

**Review by minghinmatthewlam — APPROVED** [review]
LGTM, thanks for cleaning up

**Pull request merged**